src/Passes/Source/RuleTransformationPass/RulePass.hpp

```diff
@@ -0,0 +1,245 @@
+    ///                    ├───────────────▶│   Copy and expand functions   │──┐
+    ///                    │     clone      │                               │  │
+    ///                    │   functions?   └───────────────────────────────┘  │
+    ///                    │                                │ delete dead      │
+    ///                    │                                │    code?         │
+    ///                    │                                ▼                  │
+    ///                    │                ┌───────────────────────────────┐  │
+    ///                    │                │                               │  │
+    ///                    ├───────────────▶│     Determine active code     │  │
+    ///                    │  delete dead   │                               │  │
+    ///                    │     code?      └───────────────────────────────┘  │
+    ///                    │                                │                  │
+    ///                    │                                │                  │
+    ///                    │                                ▼                  │
+    ///                    │                ┌───────────────────────────────┐  │  leave dead
+    ///                    │                │                               │  │    code?
+    ///                    │                │      Simplify phi nodes       │  │
+    ///                    │                │                               │  │
+    ///                    │                └───────────────────────────────┘  │
+    ///                    │                                │                  │
+    ///                    │                                │                  │
+    ///                    │                                ▼                  │
+    ///                    │                ┌───────────────────────────────┐  │
+    ///                    │                │                               │  │
+    ///                    │                │       Delete dead code        │  │
+    ///                    │                │                               │  │
+    ///                    │                └───────────────────────────────┘  │
+    ///                    │                                │                  │
+    ///                    │                                │                  │
+    ///                    │                                ▼                  │
+    ///                    │                ┌───────────────────────────────┐  │
+    ///                    │    fallback    │                               │  │
+    ///                    └───────────────▶│          Apply rules          │◀─┘
+    ///                                     │                               │
+    ///                                     └───────────────────────────────┘
+    ///
+    class RuleTransformationPass : public llvm::PassInfoMixin<RuleTransformationPass>
```

Comment:
Consider unifying the class name and file name.

Reply:
Done.